lasagne/layers/recurrent.py

```diff
@@ -757,3 +762,306 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
         self.cell_out = cell_out
 
         return hid_out
+
+
+class GRULayer(Layer):
+    """Gated Recurrent Layer (GRU)
+
+    Layer with gated recurrent units implemented as in [1]_, [2]_.
```

Comment:
I think they actually propose slightly different GRU equations, see footnote 1 in [2]; it's worth mentioning which one we are using.

I think we should have the update equations here, and for the other layers too, but you can ignore that here because we need to update all layers.